### PR TITLE
Tracer flare debug logs tests

### PR DESCRIPTION
**Pull request by cecile75**
## Motivation
[APMAPI-27](https://datadoghq.atlassian.net/browse/APMAPI-27) - [Java] Tracer flare debug logs

## Changes
- add check of the content of the java tracer flare
- add check of the content of the tracer.log file inside the java tracer flare

- For  these tests, I've added the function add_remote_config(self, path, payload) that uses the endpoint POST "/test/session/responses/config" in order to add the AGENT_TASK to the already existing Remote Configuration (AGENT_CONFIG).

Notes: 
- I've tested it locally on my laptop with a snapshot of the java tracer (dd-java-agent-1.41.0-SNAPSHOT)  (while removing the     "@missing_feature(context.library < "java@1.42.0", reason="config id needed to be a specific string before 1.42.0")" line of course) 

       
## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [x] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [x] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [x] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

[APMAPI-27]: https://datadoghq.atlassian.net/browse/APMAPI-27?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cbeauchesne — COMMENTED** [review]
Be careful, you modified test logic. In that situation, you must not prefix your PR title with `[java]`, otherwise, only java test are executed.

**Comment by cecile75** [comment]
> Be careful, you modified test logic. In that situation, you must not prefix your PR title with `[java]`, otherwise, only java test are executed.

I thought it was more informative than anything else. I didn't know tests were triggered or not by the PR title.

**Comment by cbeauchesne** [comment]
> I thought it was more informative than anything else. I didn't know tests were triggered or not by the PR title.

no wory, that's why we have PR reviewers 😄

**Review by cbeauchesne — APPROVED** [review]
From framework usage all good.

Though I recommend to get a review from someone familiar with the tested feature.

File: tests/parametric/conftest.py [hunk 1]
```diff
@@ -165,9 +254,13 @@ def requests(self, **kwargs) -> List[AgentRequest]:
...
         return json
 
-    def rc_requests(self):
+    def rc_requests(self, post_only=False):
         reqs = self.requests()
-        rc_reqs = [r for r in reqs if r["url"].endswith("/v0.7/config")]
+        rc_reqs = (
+            [r for r in reqs if r["url"].endswith("/v0.7/config") and r["method"] == "POST"]
+            if post_only
+            else [r for r in reqs if r["url"].endswith("/v0.7/config")]
```
Comment (mcculls) [inline tests/parametric/conftest.py:262]: would something like this also work?
```
rc_reqs = [r for r in reqs if r["url"].endswith("/v0.7/config") and (not(post_only) or r["method"] == "POST"])
```

**Pull request merged**